core/rawdb/accessors_chain.go

```diff
@@ -600,6 +600,35 @@ func ReadTdRLP(db ethdb.Reader, hash common.Hash, number uint64) rlp.RawValue {
 	return nil // Can't find the data anywhere.
 }
 
+// ReadHeadsHashes retreive's the heads hashes of the blockchain.
+func ReadHeadsHashes(db ethdb.Reader) []common.Hash {
```

Comment:
What is the purpose of "heads" compared to a single head? Shouldn't a single head exist based off of a tuple TD?

Reply:
From @mechanikalk "heads are tips in the DAG and a FIFO that we use to trim and manage state. Keep a certain depth of tips such that once the FIFO pops we trim the tip.